### ISSUE TITLE
[GCP] Cluster creation fails while executing DeployNginxIngressResources

### ISSUE THREAD

**Issue by gardener-robot-ci-1** [gardener/gardener#59]
By @ddegraf: I've tried to create a cluster on Google Cloud Platform. The creation process fails with this error message:
```
Failed to create Shoot cluster (Errors occurred during flow execution: 
'(*Botanist).DeployNginxIngressResources' returned '`.status.loadBalancer.ingress[]`
has no elements yet, i.e. external load balancer has not been created', 
'(*Botanist).WaitUntilVPNConnectionExists' returned 'timed out 
waiting for the condition') Operation will be retried.
```
Project: k8s-pm
Cluster: gcp-test

**Comment by gardener-robot-ci-1** [comment]
**Comment by [rfranzke](https://github.com/rfranzke)**
_Tuesday Jan 09, 2018 at 08:03 GMT_

----

Hi Daniel,
thanks for reporting.
The problem is that the controller manager is not allowed to create load balancers in your GCP project:

```
E0109 06:08:47.794291       1 route_controller.go:115] Couldn't reconcile node routes: error listing routes: googleapi: Error 403: Required 'compute.routes.list' permission for 'projects/sap-se-gcp-scp-core-pm', forbidden
E0109 06:08:51.503460       1 service_controller.go:698] External error while checking if load balancer "acb3e92a2f48a11e79f1b0a580a40032" exists: name, googleapi: Error 403: Required 'compute.forwardingRules.get' permission for 'projects/sap-se-gcp-scp-core-pm/regions/europe-west1/forwardingRules/acb3e92a2f48a11e79f1b0a580a40032', forbidden
```

Can you please check whether the service account you provided to the Gardener has the proper permissions, especially `Compute Admin`?

**Comment by gardener-robot-ci-1** [comment]
**Comment by [ddegraf](https://github.com/ddegraf)**
_Tuesday Jan 09, 2018 at 08:12 GMT_

----

Hi Raphael,
I've assigned the roles that are mentioned in the secret's help popup: 
![bug gcp](https://git.removed/storage/user/503/files/e4b8d78a-f51c-11e7-8b2e-132c811eac8f)
Is there any API I have to activate in the GCP Dashboard? Currently Google Compute Engine API and Google Identity and Access Management (IAM) API are the only APIs that are used.

**Comment by gardener-robot-ci-1** [comment]
**Comment by [rfranzke](https://github.com/rfranzke)**
_Tuesday Jan 09, 2018 at 08:14 GMT_

----

Okay. Not sure. I will check and come back to once I have investigated further.

**Comment by gardener-robot-ci-1** [comment]
**Comment by [D057453](https://github.com/D057453)**
_Tuesday Jan 09, 2018 at 14:51 GMT_

----

Hi,
same issue here:

> Failed to create Shoot cluster (Errors occurred during flow execution: '(*Botanist).DeployNginxIngressResources' returned '`.status.loadBalancer.ingress[]` has no elements yet, i.e. external load balancer has not been created', '(*Botanist).WaitUntilVPNConnectionExists' returned 'timed out waiting for the condition') Operation will be retried.

Project:  sap-iot-innovation
Cluster: iotsai

I'm in contact with Sebastian Stauch and Markus Liepold.

best regards,
Santo

**Comment by gardener-robot-ci-1** [comment]
**Comment by [docktofuture](https://github.com/docktofuture)**
_Wednesday Jan 10, 2018 at 11:03 GMT_

----

Add a new IAM role with member name kubeoperator@sap-cloud-platform-dev1.iam.gserviceaccount.com in your project and grant it the role "Compute Admin". This should solve the issue, due to missing permissions. 
![iam role](https://git.removed/storage/user/27689/files/1d2d3840-f5fe-11e7-90e0-1ae6b8e4628f)

**Comment by gardener-robot-ci-1** [comment]
**Comment by [ddegraf](https://github.com/ddegraf)**
_Friday Jan 12, 2018 at 13:07 GMT_

----

Adding the member works fine for me. This is just a workaround, right?

**Comment by gardener-robot-ci-1** [comment]
**Comment by [rfranzke](https://github.com/rfranzke)**
_Friday Jan 12, 2018 at 15:42 GMT_

----

Yes, hopefully. Markus Liepold and Sebastian Stauch are looking into how to correctly solve the problem.

**Comment by vlerenc** [comment]
@mliepold, @DockToFuture, @rfranzke What's the status of this ticket?

**Comment by rfranzke** [comment]
Fixed with #67
